server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -621,6 +621,7 @@ private static void updateIndexMappingsAndBuildSortOrder(IndexService indexServi
         if (!mappings.isEmpty()) {
             assert mappings.size() == 1 : mappings;
             mapperService.merge(MapperService.SINGLE_MAPPING_NAME, mappings, MergeReason.MAPPING_UPDATE);
+            mapperService.validateDynamicTemplates();
```

Comment:
This seems to only validate dynamic templates at creation time, should we move validation to MapperService#merge to make sure we cover all cases, including mapping updates that add dynamic templates on existing indices?

Reply:
Good point. I will address this.

Reply:
Dynamic template are now validated when parsed by root object type parser.
I think this has the same effect?